server/store/oci/oci.go

```diff
@@ -54,7 +52,7 @@ func New(config Config) (types.StoreService, error) {
 }
 
 func (s *store) Lookup(ctx context.Context, digest *coretypes.Digest) (*coretypes.ObjectMeta, error) {
-	_, metaReader, err := s.repository.Blobs().FetchReference(ctx, ocidigest.NewDigestFromBytes(ocidigest.SHA256, digest.Value).String())
+	_, metaReader, err := s.repository.Blobs().FetchReference(ctx, digest.ToString())
```

Comment:
i would like to prevent accidental changes to what OCI expects here, so id suggest we create another method that will return oci specific digest, and we can call `String()` method on it, for example

```go
// converts directory-specific digest into OCI digest
func ociDigest(ourDigest *core.Digest) *digest.Digest {
 // implement me
}
```
we can even move this method to the original object, but i think it may be better to keep it close to the storage implementation as not to blow up digest conversion

```go
func (digest *Digest) ToOCI() *oci.digest {
 // implement me
}
```

then on the consumer side
```suggestion
	_, metaReader, err := s.repository.Blobs().FetchReference(ctx, ociDigest(ourDigest).String())
```

Reply:
note that there are instances where we cannot convert our own digest to the digest allowed by OCI (e.g. multihash), so we may want to also return an error during conversion